### ISSUE TITLE
Slotted mixin property types are unknown in inheriting classes when running with `--verifytypes`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#2686]
**Describe the bug**
Given the following code in a module `thing.py` in a package named `foo`:

```python
from __future__ import annotations


class Mixin:
    __slots__ = ()

    def __init__(self, mixin_property: str) -> None:
        self.mixin_property: str = mixin_property


class Thing(Mixin):
    __slots__ = ('mixin_property',)

    def __init__(self, mixin_property: str) -> None:
        super().__init__(mixin_property)
```

An error is generated when running `pyright --verifytypes foo` indicating that the type is unknown for `foo.blah.Thing.mixin_property`.

**Expected behavior**
No error should be generated.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.194

**Comment by erictraut** [comment]
Why is the instance variable `mixin_property` declared in `Mixin` but it's included in the `__slots__` definition in `Thing`? Who not just included `mixin_property` in the `__slots__` definition in `Mixin`?

**Comment by bryanforbes** [comment]
In the code example I gave, that can be done. However, if `Thing` inherits from a base class and a mixin and both have non-empty `__slots__`, it will [generate a `TypeError`](https://docs.python.org/3/reference/datamodel.html#notes-on-using-slots) at runtime. Here's an example that will pass type checking (except for the second `super()` call, which I should probably file a bug report for), but fail at runtime:

```python
from __future__ import annotations


class Base:
    __slots__ = ('base_property',)

    def __init__(self, base_property: str) -> None:
        self.base_property: str = base_property


class Mixin:
    __slots__ = ('mixin_property',)

    def __init__(self, mixin_property: str) -> None:
        self.mixin_property: str = mixin_property


class Thing(Mixin, Base):
    __slots__ = ()

    def __init__(self, base_property: str, mixin_property: str) -> None:
        super().__init__(mixin_property)
        super(Mixin, self).__init__(base_property)


t = Thing('foo', mixin_property='bar')
print(t.base_property)
print(t.mixin_property)
```

To fix the `TypeError`, either `__slots__` must be removed from `Mixin` or set to `()` and `'mixin_property'` has to be added to `Thing.__slots__`.

**Comment by erictraut** [comment]
I see, thanks for the details.

The problem is that in your initial sample above, you're implicitly introducing an instance variable to the `Thing` class called `mixin_property` (via the `__slots__` assignment), but you're not providing a type for it.

You can remedy this by specifying an explicit type annotation like this:
```python
class Thing(Mixin):
    mixin_property: str
    __slots__ = ("mixin_property",)
...
```

Is that an acceptable workaround?

**Comment by bryanforbes** [comment]
It's somewhat acceptable, but it's not great since now I'm duplicating two items from `Mixin` and have to remember to keep those in sync. Duplicating `__slots__` is necessary because of Python, but it seems that duplicating the variable's type annotation shouldn't be necessary since it's technically defined in `Mixin`. If none of the classes were slotted, `--verifytypes` would report 100% type coverage, so it seems odd that in the case of `__slots__` it can't just inherit the annotation for `mixin_property` from `Mixin`. Perhaps I'm overthinking this (or underthinking it).

**Comment by erictraut** [comment]
OK, I've modified the logic to exempt `__slots__` entries from type completeness check in this circumstance. This will be included in the next release of pylance.

**Comment by erictraut** [comment]
This is included in pyright 1.1.195, which I just published.
